src/machine/image-dbus.c

```diff
@@ -56,11 +61,37 @@ int bus_image_method_remove(
         if (r == 0)
                 return 1; /* Will call us back */
 
-        r = image_remove(image);
-        if (r < 0)
+        if (pipe2(errno_pipe_fd, O_CLOEXEC|O_NONBLOCK) < 0)
+                return sd_bus_error_set_errnof(error, errno, "Failed to create pipe: %m");
+
```

Comment:
@poettering for my own knowledge, how do you feel about something like https://github.com/vcaputo/thunk/blob/master/thunk.h for streamlining the use of thunks in C for situations like this?